native/src/main/java/io/ballerina/stdlib/http/api/service/signature/converter/RecordConverter.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.api.service.signature.converter;
+
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BRefValue;
+import io.ballerina.stdlib.http.api.BallerinaConnectorException;
+import io.ballerina.stdlib.mime.util.EntityBodyHandler;
+import org.ballerinalang.langlib.value.CloneWithType;
+
+/**
+ * The record type payload converter.
+ *
+ * @since SwanLake update 1
+ */
+public class RecordConverter extends AbstractPayloadConverter {
+    Type payloadType;
+
+    public RecordConverter(Type payloadType) {
+        this.payloadType = payloadType;
+    }
+
+    @Override
+    public int getValue(BObject inRequestEntity, boolean readonly, Object[] paramFeed, int index) {
+        Object recordEntity = getRecordEntity(inRequestEntity, payloadType);
+        if (readonly && recordEntity instanceof BRefValue) {
+            ((BRefValue) recordEntity).freezeDirect();
```

Comment:
As I got to know, `freezeDirect` is for internal use. It was suggested to use `CloneReadOnly.cloneReadOnly(record);` for making this readonly. Is it possible to use it?

Reply:
@Bhashinee this clone readonly creates another clone of the record instead of using the same reference. So it can affect the memory. I also initially used it when the readonly task was doing. Then I changed it to freezeDirect as it more memory friendly.